src/Language/PureScript/Errors.hs

```diff
@@ -746,7 +744,19 @@ prettyPrintSingleError full level showWiki e = flip evalState defaultUnknownMap
             , line "They may be disallowed completely in a future version of the compiler."
             ]
     renderSimpleErrorMessage OverlappingInstances{} = internalError "OverlappingInstances: empty instance list"
-    renderSimpleErrorMessage (NoInstanceFound nm ts) =
+    renderSimpleErrorMessage (NoInstanceFound (Constraint (Qualified (Just (ModuleName [ProperName "Prim"]))
+                                                                     (ProperName "Partial"))
```

Comment:
I used a pattern synonym to address this. Generally, I think we should be using them all over the place to tidy up code, especially in the optimizer and sugar passes.